frontend/restInterfaces/BaseRestInterface.ts

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+
+/**
+ * Base Class responsible for the interaction with the server. Implements different types of HTTP-Methods.
+ */
+export abstract class BaseRestInterface {
+  /**
+   * Sends GET request to backend
+   * @param url the url
+   * @returns server response
+   * @protected
+   */
+  protected static async get<T>(url: string): Promise<T> {
```

Comment:
tsdoc for the generic parameter

Reply:
Lint does not approve if you tag it as @param, so I dont think  it is needed

Reply:
You would use [typeParam](https://typedoc.org/tags/typeParam/) for that

Reply:
lint does not know typeParam